version: '3'

services:

  redis:
    image: redis:latest
    restart: on-failure
    expose:
      - "6379"

  db:
    image: postgres:13
    environment:
      POSTGRES_DB: "cw7"
      POSTGRES_USER: "postgres"
      POSTGRES_PASSWORD: "1101"
      PGDATA: /var/lib/postgresql/data/pgdata

  app:
    build: .
    command: python manage.py migrate && python manage.py runserver 0.0.0.0:8000
    ports:
      - "8000:8000"
    depends_on:
      - db
    env_file:
      - .env

  celery:
    build: .
    tty: true
    command: celery -A config worker -l INFO  -P eventlet
    restart: on-failure
    volumes:
      - .:/app
    depends_on:
      - redis
      - db
      - app
    env_file:
      - .env

  celery-beat:
    build: .
    tty: true
    command: celery -A config beat -l INFO -P eventlet
    restart: on-failure
    volumes:
      - .:/app
    depends_on:
      - redis
      - db
      - app
    env_file:
      - .env

